src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -424,7 +424,10 @@ public EventPipeEventMetaData(PinnedStreamReader reader, int length, int fileFor
         public int MetaDataId { get; private set; }
         public string ProviderName { get; private set; }
         public string EventName { get; private set; }
+#if SUPPORT_V1_V2
         public Tuple<TypeCode, string>[] ParameterDefinitions { get; private set; }
+#endif
+        public DynamicTraceEventData Template { get; private set; }
```

Comment:
Generally speaking things associated with parsing events (and in particular templates for a TraceEvent belong in the EventPipeTraceEventParser, not here.   

If you move this parsing code to EventPipeTraceEventParser it will avoid the need for this field (which really does not belong here (in the source rather than the parser).